### PR TITLE
Ks plots without divergence lines have inner legend

### PR DESCRIPTION
**Pull request by Cecilia-Sensalari**
When the adjustment table is not available (yet), the output figure only shows the paralog Ks distribution without divergence lines. The legend is now placed inside the axis instead of outside. This is done for the "normal" mixed plot, for the anchor cluster plot, for the exp-log plot and for the log-only plot.

### CODE REVIEW

File: ksrates/fc_plotting.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def generate_mixed_plot_figure(species, x_max_lim, y_max_lim, corrected_or_not,
     if plot_correction_arrows and correction_table_available:
         fig, ax = plt.subplots(1, 1, figsize=(14.0, 7.6))
```
Comment (lohausr) [inline ksrates/fc_plotting.py:68]:
```suggestion
    if correction_table_available:
        if plot_correction_arrows:
            fig, ax = plt.subplots(1, 1, figsize=(14.0, 7.6))
        else:
            fig, ax = plt.subplots(1, 1, figsize=(14.0, 7.0))
```

Comment (lohausr) [reply ksrates/fc_plotting.py:68]: I can't make suggestions to all lines of code but this also needs to be added to the end of this function:
```
    if not correction_table_available:
        plt.tight_layout()

    return fig, ax
```

File: ksrates/fc_plotting.py [hunk 2]
```diff
@@ -480,18 +480,28 @@ def save_mixed_plot(fig_corr, fig_uncorr, ax_corr, ax_uncorr, species, paranome,
...
-                     bbox_extra_artists=(lgd, fig_corr._suptitle), bbox_inches="tight", transparent=True, format="pdf")
+    if correction_table_available:
+        ax_corr.set_position([chart_box.x0, chart_box.y0, chart_box.width*0.65, chart_box.height])
+        lgd = create_legend(ax_corr, paranome, colinearity, legend_size)
+    else:
+        lgd = ax_corr.legend(handlelength=1.5, mode="expand", loc="upper left", bbox_to_anchor=(0.54, 0.0, 0.75, 1))
+        ax_corr.set_position([chart_box.x0, chart_box.y0, chart_box.width*0.9, chart_box.height])
+
+    fig_corr.savefig(os.path.join("correction_analysis", f"{species}", f"mixed_{species}_corrected.pdf"), 
+        bbox_extra_artists=(ax_corr, lgd, fig_corr._suptitle), bbox_inches="tight", transparent=True, format="pdf")
```
Comment (lohausr) [inline ksrates/fc_plotting.py:504]: I'm not allowed to post a suggestion here directly either, but this is the new code body of this function:
```
    if correction_table_available:
        legend_size = define_legend_size(ax_corr)
        chart_box = ax_uncorr.get_position()

        ax_uncorr.set_position([chart_box.x0, chart_box.y0, chart_box.width*0.65, chart_box.height])
        lgd = create_legend(ax_uncorr, paranome, colinearity, legend_size)
        fig_uncorr.savefig(os.path.join("correction_analysis", f"{species}", f"mixed_{species}_uncorrected.pdf"),
                           bbox_extra_artists=(lgd, fig_uncorr._suptitle), bbox_inches="tight", transparent=True, 
                           format="pdf")

        ax_corr.set_position([chart_box.x0, chart_box.y0, chart_box.width*0.65, chart_box.height])
        lgd = create_legend(ax_corr, paranome, colinearity, legend_size)
        fig_corr.savefig(os.path.join("correction_analysis", f"{species}", f"mixed_{species}_corrected.pdf"),
                         bbox_extra_artists=(lgd, fig_corr._suptitle), bbox_inches="tight", transparent=True, 
                         format="pdf")
    else:
        # if not correction_table_available use a simpler layout with the legend
        # inside the plot and no right margin
        lgd = ax_uncorr.legend(handlelength=1.5, loc="upper right")
        fig_uncorr.savefig(os.path.join("correction_analysis", f"{species}", f"mixed_{species}_uncorrected.pdf"),
                           transparent=True, format="pdf")
        lgd = ax_corr.legend(handlelength=1.5, loc="upper right")
        fig_corr.savefig(os.path.join("correction_analysis", f"{species}", f"mixed_{species}_corrected.pdf"), 
                         transparent=True, format="pdf")
```

**Review by lohausr — CHANGES_REQUESTED** [review]
I have posted the changed code to remove the right margin in the comments to fc_plotting.py.

**Comment by Cecilia-Sensalari** [comment]
- matplotlib >= 3.3.3 for correct positioning of simple plot suptitle
- removed the line of code that I used to reduce the y-axis height because v3.3.3 fixes this issue

**Comment by lohausr** [comment]
Looks good to me now!

**Pull request merged**